src/main.rs

```diff
@@ -95,8 +102,8 @@ fn main() -> io::Result<()> {
         if vm.run().is_err() {
             break;
         }
-        let mut frame_buffer = [0; WIDTH * SCALE * HEIGHT * SCALE];
-        enlarge(&vm.buffer, &mut frame_buffer);
+        let mut frame_buffer = vec![0; WIDTH * scale * HEIGHT * scale];
+        enlarge(&vm.buffer, &mut frame_buffer, scale);
         window.update_with_buffer(&frame_buffer, WIDTH, HEIGHT).unwrap();
```

Comment:
remember also add scale at line 107, you need update buffer with full-size.

Reply:
Actually we used WIDTH * scale and HEIGHT * scale so many times, we can give variables to them